cli/cli-client/src/main/java/org/infinispan/cli/commands/Patch.java

```diff
@@ -0,0 +1,177 @@
+package org.infinispan.cli.commands;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.aesh.command.Command;
+import org.aesh.command.CommandDefinition;
+import org.aesh.command.CommandResult;
+import org.aesh.command.GroupCommandDefinition;
+import org.aesh.command.impl.completer.FileOptionCompleter;
+import org.aesh.command.option.Argument;
+import org.aesh.command.option.Arguments;
+import org.aesh.command.option.Option;
+import org.aesh.io.Resource;
+import org.infinispan.cli.impl.ContextAwareCommandInvocation;
+import org.infinispan.cli.logging.Messages;
+import org.infinispan.cli.patching.PatchTool;
+import org.kohsuke.MetaInfServices;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 11.0
+ **/
+@MetaInfServices(Command.class)
+@GroupCommandDefinition(name = Patch.CMD, description = "Patch operations", groupCommands = {Patch.Create.class, Patch.Describe.class, Patch.Install.class, Patch.Ls.class, Patch.Rollback.class})
+public class Patch extends CliCommand {
+
+   public static final String CMD = "patch";
+
+   @Override
+   public CommandResult exec(ContextAwareCommandInvocation commandInvocation) {
+      // This command serves only to wrap the sub-commands
+      return CommandResult.FAILURE;
+   }
+
+   @CommandDefinition(name = Create.CMD, description = "Creates a patch archive")
+   public static class Create extends CliCommand {
+      public static final String CMD = "create";
+
+      @Option(defaultValue = "", shortName = 'q', description = "A qualifier for this patch (e.g. `one-off`)")
+      String qualifier;
+
+      @Arguments(completer = FileOptionCompleter.class, description = "The path to the patch archive, the path to the target server and one or more paths to the source servers")
+      List<Resource> paths;
+
+      @Override
+      public CommandResult exec(ContextAwareCommandInvocation invocation) {
+         if (paths.size() < 3) {
+            throw Messages.MSG.patchCreateArgumentsRequired();
+         }
+         PatchTool patchTool = new PatchTool(invocation.getShellOutput(), invocation.getShellError());
+         try {
+            Path patch = Paths.get(paths.get(0).getAbsolutePath());
+            Path target = Paths.get(paths.get(1).getAbsolutePath());
+            Path sources[] = new Path[paths.size() - 2];
+            for (int i = 2; i < paths.size(); i++) {
+               sources[i - 2] = Paths.get(paths.get(i).getAbsolutePath());
+            }
+            patchTool.createPatch(qualifier, patch, target, sources);
+            return CommandResult.SUCCESS;
+         } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
exception handling is missing in all subcommands

Reply:
Fixed